packages/yarnpkg-core/sources/scriptUtils.ts

```diff
@@ -142,22 +151,24 @@ export async function makeScriptEnv({project, locator, binFolder, lifecycleScrip
 const MAX_PREPARE_CONCURRENCY = 2;
 const prepareLimit = pLimit(MAX_PREPARE_CONCURRENCY);
 
-export async function prepareExternalProject(cwd: PortablePath, outputPath: PortablePath, {configuration, report, workspace = null}: {configuration: Configuration, report: Report, workspace?: string | null}) {
+export async function prepareExternalProject(cwd: PortablePath, outputPath: PortablePath, {configuration, report, workspace = null, locator}: {configuration: Configuration, report: Report, workspace?: string | null, locator: Locator}) {
```

Comment:
This is a breaking change btw

Reply:
What would be the best way to make this non-breaking? The only option I can see is to default to `null` and not output the name when it's not supplied (which isn't the worst thing in the world)

Reply:
Yep, that sounds reasonable 👍 (we currently accept breaking change, but in this case I think making this parameter optional is a good idea anyway)

Reply:
Updated